src/main/java/com/commercetools/sync/inventories/InventorySync.java

```diff
@@ -0,0 +1,444 @@
+                  final TypeService typeService) {
+        super(new InventorySyncStatistics(), syncOptions);
+        this.inventoryService = inventoryService;
+        this.typeService = typeService;
+    }
+
+    /**
+     * {@inheritDoc}
+     *
+     * <p>
+     *     <strong>NOTE:</strong> {@code inventoryDrafts} are compared with existing inventory entries by {@code sku}
+     *     and {@code supplyChannel} key. Every {@link InventoryEntryDraft} that contains {@code supplyChannel} should
+     *     either:
+     *     <ul>
+     *         <li>have {@code supplyChannel} expanded, that means
+     *         {@code inventoryEntryDraft.getSupplyChannel().getObj()} should return {@link Channel} object,
+     *         which contains channel key</li>
+     *         <li>or have {@code supplyChannel} not expanded and {@code supplyChannel} key should be provided in
+     *         place of reference id, that means {@code inventoryEntryDraft.getSupplyChannel().getObj()} should
+     *         return {@code null} and {@code inventoryEntryDraft.getSupplyChannel().getId()} should
+     *         return {@code supplyChannel} key</li>
+     *     </ul>
+     *     This is important for proper resources comparision.
+     * </p>
+     */
+    @Override
+    public CompletionStage<InventorySyncStatistics> syncDrafts(@Nonnull final List<InventoryEntryDraft>
+                                                                       inventoryDrafts) {
+        return super.syncDrafts(inventoryDrafts);
+    }
+
+    /**
+     * Performs full process of synchronisation between inventory entries present in a system
+     * and passed {@code inventories}. This is accomplished by:
+     * <ul>
+     *     <li>Comparing entries and drafts by {@code sku} and {@code supplyChannel} key</li>
+     *     <li>Calculating of necessary updates and creation commands</li>
+     *     <li>Actually <strong>performing</strong> changes in a target CTP project</li>
+     * </ul>
+     * The process is customized according to {@link InventorySyncOptions} passed to constructor of this object.
+     *
+     * <p><strong>Inherited doc:</strong>
+     * {@inheritDoc}
+     *
+     * @param inventories {@link List} of {@link InventoryEntryDraft} resources that would be synced into CTP project.
+     * @return {@link CompletionStage} with {@link InventorySyncStatistics} holding statistics of all sync
+     *                                           processes performed by this sync instance
+     */
+    @Override
+    protected CompletionStage<InventorySyncStatistics> processDrafts(@Nonnull final List<InventoryEntryDraft>
+                                                                             inventories) {
+        return populateSupplyChannels(inventories)
+                .thenCompose(v -> splitToBatchesAndProcess(inventories))
+                .exceptionally(ex -> {
+                    handleFailure("Failed to find supply channels", ex);
```

Comment:
Exception handling will be adjusted further with https://github.com/commercetools/commercetools-sync-java/issues/31 and https://github.com/commercetools/commercetools-sync-java/issues/30 issues